bootstrap-salt.sh

```diff
@@ -3019,6 +3019,11 @@ __install_saltstack_debian_repository() {
 
     __PACKAGES=''
 
+    # Install procps packages on systems missing it, e.g. debian:stretch-slim container images
+    if ! [ -x "$(command -v ps)" ]; then
+        __PACKAGES="${__PACKAGES} procps"
```

Comment:
We already have the code, that ensures `procps` package is going to be installed for Debian bootstraps at:
https://github.com/saltstack/salt-bootstrap/blob/6e1b09c8afb7a7a487a5f8650a1e0ace760f9082/bootstrap-salt.sh#L3078

It should be added to the packages list for both `stable` and `git` modes.
Do you have a particular issue with it?

Reply:
I'm confused. I've downloaded the Version 2018.08.15 as suggested. And then the docker build ouput was:

```
Step 8/9 : RUN /tmp/bootstrap-salt.sh -x python$PYTHON_VERSION $BOOTSTRAP_OPTS $SALT_VERSION
 ---> Running in 8ef0c9fea017
/tmp/bootstrap-salt.sh: 547: /tmp/bootstrap-salt.sh: ps: not found
 *  INFO: Running version: 2018.08.15
 *  INFO: Executed by:
 *  INFO: Command line: '/tmp/bootstrap-salt.sh -x python3 -U stable'
 *  INFO: Detected -x option. Using python3 to install Salt.
```

Downloading now via _curl -o bootstrap-salt.sh -L https://bootstrap.saltstack.com_ the script has many changes and works as aspected.

Sorry for that.

Reply:
@pluhmen The script uses `ps` early to determine which command it was called by.
https://github.com/saltstack/salt-bootstrap/blob/6e1b09c8afb7a7a487a5f8650a1e0ace760f9082/bootstrap-salt.sh#L552

That doesn't affect anything and used for debugging purposes only. The error message looks deceptive though.

Reply:
In depth: I've downloaded the script with _curl -o bootstrap-salt.sh -L https://bootstrap.saltstack.com_ and get the latestest stable version 2018.08.15 with sha256 6d414a39439a7335af1b78203f9d37e11c972b3c49c519742c6405e2944c6c4b

Running the script in a debian:stretch-slim container outputs the following.

``` 
bash -x /tmp/bootstrap-salt.sh -x python3
...
+ printf '\033[1;32m *  INFO\033[0m: %s\n' 'Running install_debian_deps()'
 *  INFO: Running install_debian_deps()
+ install_debian_deps
+ '[' 1 -eq 0 ']'
+ export DEBIAN_FRONTEND=noninteractive
+ DEBIAN_FRONTEND=noninteractive
+ __wait_for_apt
+ echodebug 'Checking if apt process is currently running.'
+ '[' 0 -eq 1 ']'
+ WAIT_TIMEOUT=900
+ ps -C apt,apt-get,aptitude,dpkg
./bootstrap-salt.sh: line 1805: ps: command not found
+ echodebug 'No apt processes are currently running.'
+ '[' 0 -eq 1 ']'
+ apt-get update
Hit:1 http://security-cdn.debian.org/debian-security stretch/updates InRelease
Ign:2 http://cdn-fastly.deb.debian.org/debian stretch InRelease
Hit:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease
^C++ __exit_cleanup
++ EXIT_CODE=130
++ '[' stable = git ']'
++ '[' -p /tmp/bootstrap-salt.logpipe ']'
++ echodebug 'Removing the logging pipe /tmp/bootstrap-salt.logpipe'
++ '[' 0 -eq 1 ']'
++ rm -f /tmp/bootstrap-salt.logpipe
+++ ps ax
./bootstrap-salt.sh: line 521: ps: command not found
+++ grep tee
+++ grep /tmp/bootstrap-salt.log
+++ awk '{print $1}'
++ TEE_PID=
++ '[' '' = '' ']'
++ exit 130
+ __exit_cleanup
+ EXIT_CODE=130
+ '[' stable = git ']'
+ '[' -p /tmp/bootstrap-salt.logpipe ']'
++ ps ax
./bootstrap-salt.sh: line 521: ps: command not found
++ grep /tmp/bootstrap-salt.log
++ grep tee
++ awk '{print $1}'
+ TEE_PID=
+ '[' '' = '' ']'
+ exit 130
```
Before the installation of procps package is triggerd, the apt process is determined with ps.

At least https://github.com/saltstack/salt-bootstrap/commit/45c125c32611dfac812aedde0c0507e587319bbe works without ps errors.

Reply:
Fine. Thanks!

Reply:
are you okay to close this @pluhmen since this is fixed in develop?

Reply:
@Ch3LL Yes, can be closed.